## Selección, limpieza y alistamiento de datos

In [32]:
import pandas as pd
import numpy as np

In [33]:
#Initial view of the data
raw_data = pd.read_csv("../DataAWS.csv")
raw_data.head()

,periodo,estu_tipodocumento,estu_consecutivo,cole_area_ubicacion,cole_bilingue,cole_calendario,cole_caracter,cole_cod_dane_establecimiento,cole_cod_dane_sede,cole_cod_depto_ubicacion,...,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,desemp_ingles,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
0,20162,TI,SB11201620604627,URBANO,N,A,TÉCNICO/ACADÉMICO,115238000698,115238000698,15,...,NaN,NaN,NaN,A2,61.00,59.00,63.0,57.0,56.0,295.0
1,20132,TI,SB11201320007175,URBANO,N,A,TÉCNICO/ACADÉMICO,315762000076,315762000076,15,...,Si,Si,No,A1,43.00,52.00,NaN,NaN,NaN,NaN
2,20102,CC,SB11201020409042,URBANO,N,A,TÉCNICO,115790000031,115790000031,15,...,Si,Si,Si,A-,38.53,45.38,NaN,NaN,NaN,NaN
3,20142,TI,SB11201420116599,RURAL,N,A,TÉCNICO/ACADÉMICO,215759000524,215759000524,15,...,Si,No,No,A1,52.00,55.00,54.0,60.0,54.0,277.0
4,20152,TI,SB11201520349379,URBANO,N,A,TÉCNICO,115835000283,115835000283,15,...,No,No,Si,A1,55.00,52.00,50.0,53.0,47.0,254.0


In [34]:
raw_data.shape

(215234, 51)

In [35]:
#Colums names
raw_data.columns

Index(['periodo', 'estu_tipodocumento', 'estu_consecutivo',
       'cole_area_ubicacion', 'cole_bilingue', 'cole_calendario',
       'cole_caracter', 'cole_cod_dane_establecimiento', 'cole_cod_dane_sede',
       'cole_cod_depto_ubicacion', 'cole_cod_mcpio_ubicacion',
       'cole_codigo_icfes', 'cole_depto_ubicacion', 'cole_genero',
       'cole_jornada', 'cole_mcpio_ubicacion', 'cole_naturaleza',
       'cole_nombre_establecimiento', 'cole_nombre_sede',
       'cole_sede_principal', 'estu_cod_depto_presentacion',
       'estu_cod_mcpio_presentacion', 'estu_cod_reside_depto',
       'estu_cod_reside_mcpio', 'estu_depto_presentacion', 'estu_depto_reside',
       'estu_estadoinvestigacion', 'estu_estudiante', 'estu_fechanacimiento',
       'estu_genero', 'estu_mcpio_presentacion', 'estu_mcpio_reside',
       'estu_nacionalidad', 'estu_pais_reside', 'estu_privado_libertad',
       'fami_cuartoshogar', 'fami_educacionmadre', 'fami_educacionpadre',
       'fami_estratovivienda', 'fami_perso

In [36]:
#Nan values
raw_data.isna().sum()

periodo                              0
estu_tipodocumento                   0
estu_consecutivo                     0
cole_area_ubicacion                 80
cole_bilingue                    17277
cole_calendario                     49
cole_caracter                      980
cole_cod_dane_establecimiento        0
cole_cod_dane_sede                   0
cole_cod_depto_ubicacion             0
cole_cod_mcpio_ubicacion             0
cole_codigo_icfes                 1616
cole_depto_ubicacion                 0
cole_genero                          0
cole_jornada                         0
cole_mcpio_ubicacion                 0
cole_naturaleza                      0
cole_nombre_establecimiento          0
cole_nombre_sede                     0
cole_sede_principal                  0
estu_cod_depto_presentacion          0
estu_cod_mcpio_presentacion          0
estu_cod_reside_depto             1234
estu_cod_reside_mcpio             1234
estu_depto_presentacion              0
estu_depto_reside        

In [37]:
raw_data.isna().sum(axis=0)/raw_data.shape[0]*100

periodo                           0.000000
estu_tipodocumento                0.000000
estu_consecutivo                  0.000000
cole_area_ubicacion               0.037169
cole_bilingue                     8.027078
cole_calendario                   0.022766
cole_caracter                     0.455318
cole_cod_dane_establecimiento     0.000000
cole_cod_dane_sede                0.000000
cole_cod_depto_ubicacion          0.000000
cole_cod_mcpio_ubicacion          0.000000
cole_codigo_icfes                 0.750811
cole_depto_ubicacion              0.000000
cole_genero                       0.000000
cole_jornada                      0.000000
cole_mcpio_ubicacion              0.000000
cole_naturaleza                   0.000000
cole_nombre_establecimiento       0.000000
cole_nombre_sede                  0.000000
cole_sede_principal               0.000000
estu_cod_depto_presentacion       0.000000
estu_cod_mcpio_presentacion       0.000000
estu_cod_reside_depto             0.573329
estu_cod_re

In [38]:
raw_data["periodo"].unique()

array([20162, 20132, 20102, 20142, 20152, 20172, 20194, 20122, 20112,
       20141, 20224, 20111, 20201, 20101, 20151, 20121, 20131, 20181,
       20211, 20221])

In [39]:
cols = ["punt_lectura_critica","punt_c_naturales","punt_sociales_ciudadanas","punt_ingles","punt_matematicas","punt_global"]
raw_data.groupby("periodo")[cols].count()


,punt_lectura_critica,punt_c_naturales,punt_sociales_ciudadanas,punt_ingles,punt_matematicas,punt_global
periodo,,,,,,
20101,0,0,0,658,658,0
20102,0,0,0,20202,20202,0
20111,0,0,0,487,487,0
20112,0,0,0,20235,20235,0
20121,0,0,0,460,460,0
20122,0,0,0,22352,22352,0
20131,0,0,0,340,340,0
20132,0,0,0,18109,18109,0
20141,0,0,0,307,307,0


no hay registros para puntajes antes de 20141, por lo que se eliminan estos registros, no son relevantes ni aportan valor en los análisis.

In [40]:
raw_data1 = raw_data[raw_data["periodo"] >= 20142]
raw_data1.shape

(132084, 51)

In [41]:
raw_data1.groupby("periodo")[cols].count()

,punt_lectura_critica,punt_c_naturales,punt_sociales_ciudadanas,punt_ingles,punt_matematicas,punt_global
periodo,,,,,,
20142,17325,17325,17325,17325,17325,17325
20151,299,299,299,299,299,299
20152,16538,16538,16538,16538,16538,16538
20162,16840,16840,16840,16840,16840,16840
20172,16872,16872,16872,16872,16872,16872
20181,87,87,87,87,87,87
20194,33474,33474,33474,33474,33474,33474
20201,94,94,94,94,94,94
20211,65,65,65,64,65,65


In [42]:
raw_data1.isna().sum(axis=0)/raw_data.shape[0]*100

periodo                          0.000000
estu_tipodocumento               0.000000
estu_consecutivo                 0.000000
cole_area_ubicacion              0.000000
cole_bilingue                    5.369969
cole_calendario                  0.000000
cole_caracter                    0.369830
cole_cod_dane_establecimiento    0.000000
cole_cod_dane_sede               0.000000
cole_cod_depto_ubicacion         0.000000
cole_cod_mcpio_ubicacion         0.000000
cole_codigo_icfes                0.000000
cole_depto_ubicacion             0.000000
cole_genero                      0.000000
cole_jornada                     0.000000
cole_mcpio_ubicacion             0.000000
cole_naturaleza                  0.000000
cole_nombre_establecimiento      0.000000
cole_nombre_sede                 0.000000
cole_sede_principal              0.000000
estu_cod_depto_presentacion      0.000000
estu_cod_mcpio_presentacion      0.000000
estu_cod_reside_depto            0.028806
estu_cod_reside_mcpio            0

# Antes de droppear columnas

Antes de elegir columnas a droppear se analizan como son los valores que están dentro de columnas numéricas y categóricas

Numéricas

In [43]:
raw_data1.describe()

,periodo,cole_cod_dane_establecimiento,cole_cod_dane_sede,cole_cod_depto_ubicacion,cole_cod_mcpio_ubicacion,cole_codigo_icfes,estu_cod_depto_presentacion,estu_cod_mcpio_presentacion,estu_cod_reside_depto,estu_cod_reside_mcpio,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
count,132084.000000,1.320840e+05,1.320840e+05,132084.0,132084.000000,132084.000000,132084.000000,132084.000000,132022.000000,132022.000000,132081.000000,132084.000000,132084.000000,132084.000000,132084.000000,132084.000000
mean,20181.860415,1.915744e+11,1.937988e+11,15.0,15410.978870,131007.155575,15.738242,16105.002771,15.102581,15506.584986,51.659785,54.356743,51.333053,52.838126,53.735388,264.789104
std,28.855334,9.073463e+10,9.180739e+10,0.0,288.718222,210381.867718,6.614105,6610.937236,2.192938,2210.896201,10.940840,11.219249,11.095774,9.630691,9.751058,46.214481
min,20142.000000,1.150010e+11,1.150010e+11,15.0,15001.000000,4747.000000,5.000000,5001.000000,5.000000,5001.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,20152.000000,1.153320e+11,1.153670e+11,15.0,15176.000000,5595.000000,15.000000,15176.000000,15.000000,15176.000000,44.000000,47.000000,43.000000,46.000000,47.000000,231.000000
50%,20172.000000,1.158040e+11,1.158140e+11,15.0,15407.000000,47316.000000,15.000000,15299.000000,15.000000,15403.000000,51.000000,54.000000,51.500000,53.000000,54.000000,263.000000
75%,20194.000000,3.150010e+11,3.150010e+11,15.0,15740.000000,132019.000000,15.000000,15753.000000,15.000000,15720.000000,58.000000,62.000000,59.000000,60.000000,61.000000,297.000000
max,20224.000000,4.158390e+11,5.154690e+11,15.0,15897.000000,783480.000000,99.000000,99001.000000,85.000000,85225.000000,100.000000,100.000000,100.000000,100.000000,100.000000,471.000000


Categóricas

In [44]:
cat_cols = raw_data1.select_dtypes(exclude=["number"]).columns
for col in cat_cols:
    print(raw_data1[col].value_counts(dropna=False))

estu_tipodocumento
TI     112001
CC      18525
CR       1361
NES        82
CE         56
PEP        44
PPT         6
PE          5
RC          2
V           1
PC          1
Name: count, dtype: int64
estu_consecutivo
SB11202240270178    4
SB11202240312312    4
SB11202240319527    4
SB11202240156027    4
SB11202240348980    4
                   ..
SB11201520289631    1
SB11201720064109    1
SB11201720077724    1
SB11201520363660    1
SB11201520282928    1
Name: count, Length: 100049, dtype: int64
cole_area_ubicacion
URBANO    111377
RURAL      20707
Name: count, dtype: int64
cole_bilingue
N      118966
NaN     11558
S        1560
Name: count, dtype: int64
cole_calendario
A    132084
Name: count, dtype: int64
cole_caracter
TÉCNICO/ACADÉMICO    59724
ACADÉMICO            42704
TÉCNICO              27831
NO APLICA             1029
NaN                    796
Name: count, dtype: int64
cole_depto_ubicacion
BOYACA    132084
Name: count, dtype: int64
cole_genero
MIXTO        129047
FEMENINO     

quitar duplicados considerando que la variable estu_consecutivo debería ser única por registro

In [45]:
raw_data2 = raw_data1.drop_duplicates(subset=['estu_consecutivo'], keep="first")
raw_data2.shape

(100049, 51)

# Drop inicial de columnas que no son de interes

In [46]:
initial_cols_drop = ['estu_tipodocumento', 'estu_consecutivo',
       'cole_cod_dane_establecimiento', 'cole_cod_dane_sede',
       'cole_cod_depto_ubicacion',
       'cole_codigo_icfes', 'cole_depto_ubicacion', 'cole_genero',
       'cole_nombre_sede', 'cole_sede_principal', 'estu_cod_depto_presentacion',
       'estu_cod_mcpio_presentacion', 'estu_cod_reside_depto',
       'estu_cod_reside_mcpio', 'estu_depto_presentacion', 'estu_depto_reside',
       'estu_estadoinvestigacion', 'estu_estudiante', 'estu_fechanacimiento', 'estu_mcpio_presentacion', 'estu_mcpio_reside',
       'estu_nacionalidad', 'estu_pais_reside', 'desemp_ingles']

raw_data3 = raw_data2.drop(columns=initial_cols_drop, errors="ignore")
raw_data3.shape

(100049, 27)

In [47]:
raw_data3.describe()

,periodo,cole_cod_mcpio_ubicacion,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
count,100049.000000,100049.000000,100047.000000,100049.000000,100049.000000,100049.000000,100049.000000,100049.000000
mean,20173.386251,15409.983898,51.631913,54.065318,51.694410,53.097462,53.374377,264.741327
std,27.041575,288.145931,10.681405,11.237403,10.893389,9.543411,9.676324,45.688126
min,20142.000000,15001.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,20152.000000,15176.000000,44.000000,46.000000,44.000000,46.000000,47.000000,232.000000
50%,20162.000000,15407.000000,50.000000,54.000000,52.000000,53.000000,53.000000,263.000000
75%,20194.000000,15740.000000,58.000000,62.000000,59.000000,60.000000,60.000000,296.000000
max,20224.000000,15897.000000,100.000000,100.000000,100.000000,100.000000,100.000000,471.000000


de los puntajes de ingles seguimos con 2 nulos, se inputarán usando los scores de las otras áreas así como el puntaje global

In [48]:
missing_rows = raw_data3[raw_data3["punt_ingles"].isna()]
missing_rows


,periodo,cole_area_ubicacion,cole_bilingue,cole_calendario,cole_caracter,cole_cod_mcpio_ubicacion,cole_jornada,cole_mcpio_ubicacion,cole_naturaleza,cole_nombre_establecimiento,...,fami_tieneautomovil,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
9092,20211,URBANO,N,A,TÉCNICO/ACADÉMICO,15001,MAÑANA,TUNJA,NO OFICIAL,COLEGIO CORPORACION SANTO DOMINGO,...,Si,Si,Si,Si,NaN,29.0,27.0,31.0,36.0,154.0
89246,20224,URBANO,N,A,TÉCNICO/ACADÉMICO,15332,UNICA,GÜICÁN DE LA SIERRA,OFICIAL,I.E. NORMAL SUPERIOR NUESTRA SEÑORA DEL ROSARIO,...,No,Si,No,No,NaN,37.0,28.0,41.0,39.0,181.0


In [49]:
raw_data3.loc[raw_data3["punt_ingles"].isna(), "punt_ingles"] = 13/5*raw_data3["punt_global"] - 3*raw_data3["punt_lectura_critica"] 
- 3*raw_data3["punt_c_naturales"] - 3*raw_data3["punt_sociales_ciudadanas"] - 3*raw_data3["punt_matematicas"]
raw_data3[raw_data3["punt_ingles"].isna()]

,periodo,cole_area_ubicacion,cole_bilingue,cole_calendario,cole_caracter,cole_cod_mcpio_ubicacion,cole_jornada,cole_mcpio_ubicacion,cole_naturaleza,cole_nombre_establecimiento,...,fami_tieneautomovil,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global


In [50]:
cat_cols = raw_data3.select_dtypes(exclude=["number"]).columns
for col in cat_cols:
    print(raw_data3[col].value_counts(dropna=False))

cole_area_ubicacion
URBANO    84547
RURAL     15502
Name: count, dtype: int64
cole_bilingue
N      90396
NaN     8531
S       1122
Name: count, dtype: int64
cole_calendario
A    100049
Name: count, dtype: int64
cole_caracter
TÉCNICO/ACADÉMICO    45502
ACADÉMICO            32206
TÉCNICO              21246
NO APLICA              646
NaN                    449
Name: count, dtype: int64
cole_jornada
COMPLETA    48411
MAÑANA      34599
UNICA        6354
TARDE        4745
SABATINA     4171
NOCHE        1769
Name: count, dtype: int64
cole_mcpio_ubicacion
TUNJA           15568
DUITAMA         11998
SOGAMOSO        11684
CHIQUINQUIRA     4744
PAIPA            2658
                ...  
TUNUNGUÁ           40
CHIVATÁ            38
PISBA              27
BUSBANZÁ           13
BUSBANZA           12
Name: count, Length: 174, dtype: int64
cole_naturaleza
OFICIAL       81513
NO OFICIAL    18536
Name: count, dtype: int64
cole_nombre_establecimiento
COLEGIO GUILLERMO LEON VALENCIA                        

# Imputaciones variables categóricas

In [51]:
#caracter del colegio
raw_data3.loc[raw_data3["cole_caracter"].isna(), "cole_caracter"] = "NO REPORTADO"

Luego de analizar un poco las preguntas planteadas no encontramos relevancia en mantener las columnas relacionadas con el entorno socioeconómico, por lo que decidimos droppear esas columnas

Así como otras columnas como género y calendario

In [52]:
cols_drop2 = [
        'fami_tieneautomovil',
        'cole_calendario', ]
raw_data4 = raw_data3.drop(columns=cols_drop2, errors="ignore")
raw_data4.shape

(100049, 25)

In [53]:
cat_cols = raw_data4.select_dtypes(exclude=["number"]).columns
for col in cat_cols:
    print(raw_data4[col].value_counts(dropna=False))

cole_area_ubicacion
URBANO    84547
RURAL     15502
Name: count, dtype: int64
cole_bilingue
N      90396
NaN     8531
S       1122
Name: count, dtype: int64
cole_caracter
TÉCNICO/ACADÉMICO    45502
ACADÉMICO            32206
TÉCNICO              21246
NO APLICA              646
NO REPORTADO           449
Name: count, dtype: int64
cole_jornada
COMPLETA    48411
MAÑANA      34599
UNICA        6354
TARDE        4745
SABATINA     4171
NOCHE        1769
Name: count, dtype: int64
cole_mcpio_ubicacion
TUNJA           15568
DUITAMA         11998
SOGAMOSO        11684
CHIQUINQUIRA     4744
PAIPA            2658
                ...  
TUNUNGUÁ           40
CHIVATÁ            38
PISBA              27
BUSBANZÁ           13
BUSBANZA           12
Name: count, Length: 174, dtype: int64
cole_naturaleza
OFICIAL       81513
NO OFICIAL    18536
Name: count, dtype: int64
cole_nombre_establecimiento
COLEGIO GUILLERMO LEON VALENCIA                              2308
COLEGIO DE BOYACA                          

volvemos a revisar donde tenemos nulos

In [54]:
raw_data4.isna().sum(axis=0)/raw_data4.shape[0]*100

periodo                        0.000000
cole_area_ubicacion            0.000000
cole_bilingue                  8.526822
cole_caracter                  0.000000
cole_cod_mcpio_ubicacion       0.000000
cole_jornada                   0.000000
cole_mcpio_ubicacion           0.000000
cole_naturaleza                0.000000
cole_nombre_establecimiento    0.000000
estu_genero                    0.114944
estu_privado_libertad          0.000000
fami_cuartoshogar              1.540245
fami_educacionmadre            2.380833
fami_educacionpadre            2.379834
fami_estratovivienda           2.640706
fami_personashogar             1.597217
fami_tienecomputador           1.488271
fami_tieneinternet             2.462793
fami_tienelavadora             1.490270
punt_ingles                    0.000000
punt_matematicas               0.000000
punt_sociales_ciudadanas       0.000000
punt_c_naturales               0.000000
punt_lectura_critica           0.000000
punt_global                    0.000000


Imputación final para colegios bilingues

In [55]:
raw_data4.loc[raw_data4["cole_bilingue"].isna(), "cole_bilingue"] = "NO REPORTADO"


In [56]:
raw_data4.head()

,periodo,cole_area_ubicacion,cole_bilingue,cole_caracter,cole_cod_mcpio_ubicacion,cole_jornada,cole_mcpio_ubicacion,cole_naturaleza,cole_nombre_establecimiento,estu_genero,...,fami_personashogar,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
0,20162,URBANO,N,TÉCNICO/ACADÉMICO,15238,MAÑANA,DUITAMA,OFICIAL,COLEGIO GUILLERMO LEON VALENCIA,M,...,NaN,NaN,NaN,NaN,61.0,59.0,63.0,57.0,56.0,295.0
3,20142,RURAL,N,TÉCNICO/ACADÉMICO,15759,MAÑANA,SOGAMOSO,OFICIAL,INSTITUCION EDUCATIVA LA INDEPENDENCIA,F,...,Seis,Si,No,No,52.0,55.0,54.0,60.0,54.0,277.0
4,20152,URBANO,N,TÉCNICO,15835,COMPLETA,TURMEQUE,OFICIAL,I.E. TECNICA INDUSTRIAL,F,...,Tres,No,No,Si,55.0,52.0,50.0,53.0,47.0,254.0
7,20172,URBANO,N,TÉCNICO/ACADÉMICO,15176,MAÑANA,CHIQUINQUIRA,OFICIAL,I.E. TECNICA PIO ALBERTO FERRO PEÑA,F,...,5 a 6,No,No,Si,59.0,58.0,46.0,51.0,45.0,253.0
8,20172,URBANO,N,TÉCNICO/ACADÉMICO,15759,MAÑANA,SOGAMOSO,NO OFICIAL,COLEGIO COOPERATIVO REYES PATRIA,M,...,3 a 4,Si,Si,Si,75.0,66.0,70.0,65.0,64.0,335.0


Nuevo encoding 

In [57]:

mappings = {
    "periodo": lambda x: int(x) if pd.notna(x) else np.nan,

    "cole_area_ubicacion": {
        "RURAL": 0,
        "URBANO": 1,
    },
    "cole_bilingue": {
        "N": 0,
        "S": 1,
        "NO REPORTADO": -1,
    },
    "cole_caracter": {
        "TÉCNICO": 1,
        "TÉCNICO/ACADÉMICO": 2,
        "ACADÉMICO": 3,
        "NO APLICA": 4,
        "NO REPORTADO": -1,
    },
    "cole_jornada": {
        "MAÑANA": 1,
        "TARDE": 2,
        "NOCHE": 3,
        "SABATINA": 4,
        "UNICA": 5,
        "COMPLETA": 6,
    },
    "cole_naturaleza": {
        "NO OFICIAL": 0,
        "OFICIAL": 1,
    },
    "estu_genero": {
        "F": 0,
        "M": 1,
    },
    "estu_privado_libertad": {
        "S": 0,
        "N": 1,
    },

    # Ordinal variables
    "fami_cuartoshogar": {
        "Uno": 1,
        "Dos": 2,
        "Tres": 3,
        "Cuatro": 4,
        "Cinco": 5,
        "Seis": 6,
        "Seis o mas": 6,
        "Siete": 7,
        "Ocho": 8,
        "Nueve": 9,
        "Diez o más": 10,
    },
    "fami_educacionmadre": {
        "Ninguno": 0,
        "Primaria incompleta": 1,
        "Primaria completa": 2,
        "Secundaria (Bachillerato) incompleta": 3,
        "Secundaria (Bachillerato) completa": 4,
        "Técnica o tecnológica incompleta": 5,
        "Técnica o tecnológica completa": 6,
        "Educación profesional incompleta": 7,
        "Educación profesional completa": 8,
        "Postgrado": 9,
        "No sabe": -1,
        "No Aplica": -1,
    },
    "fami_educacionpadre": {
        "Ninguno": 0,
        "Primaria incompleta": 1,
        "Primaria completa": 2,
        "Secundaria (Bachillerato) incompleta": 3,
        "Secundaria (Bachillerato) completa": 4,
        "Técnica o tecnológica incompleta": 5,
        "Técnica o tecnológica completa": 6,
        "Educación profesional incompleta": 7,
        "Educación profesional completa": 8,
        "Postgrado": 9,
        "No sabe": -1,
        "No Aplica": -1,
    },
    "fami_estratovivienda": {
        "Sin Estrato": 0,
        "Estrato 1": 1,
        "Estrato 2": 2,
        "Estrato 3": 3,
        "Estrato 4": 4,
        "Estrato 5": 5,
        "Estrato 6": 6,
    },
    "fami_personashogar": {
        "Una": 1,
        "Dos": 2,
        "Tres": 3,
        "Cuatro": 4,
        "Cinco": 5,
        "Seis": 6,
        "Siete": 7,
        "Ocho": 8,
        "Nueve": 9,
        "Diez": 10,
        "Once": 11,
        "1 a 2": 1.5,
        "3 a 4": 3.5,
        "5 a 6": 5.5,
        "7 a 8": 7.5,
        "9 o más": 9.5,
        "Doce o más": 12,
    },

    # Binary / nominal
    "fami_tienecomputador": {
        "No": 0,
        "Si": 1,
    },
    "fami_tieneinternet": {
        "No": 0,
        "Si": 1,
    },
    "fami_tienelavadora": {
        "No": 0,
        "Si": 1,
    },
}

def encode_series(series, mapper):
    if callable(mapper):
        return series.apply(mapper)
    return series.map(mapper)

data_encoded = raw_data4.copy()

for col, mapper in mappings.items():
    data_encoded[col] = encode_series(data_encoded[col], mapper)

data_encoded.head()

,periodo,cole_area_ubicacion,cole_bilingue,cole_caracter,cole_cod_mcpio_ubicacion,cole_jornada,cole_mcpio_ubicacion,cole_naturaleza,cole_nombre_establecimiento,estu_genero,...,fami_personashogar,fami_tienecomputador,fami_tieneinternet,fami_tienelavadora,punt_ingles,punt_matematicas,punt_sociales_ciudadanas,punt_c_naturales,punt_lectura_critica,punt_global
0,20162,1,0,2,15238,1,DUITAMA,1,COLEGIO GUILLERMO LEON VALENCIA,1.0,...,NaN,NaN,NaN,NaN,61.0,59.0,63.0,57.0,56.0,295.0
3,20142,0,0,2,15759,1,SOGAMOSO,1,INSTITUCION EDUCATIVA LA INDEPENDENCIA,0.0,...,6.0,1.0,0.0,0.0,52.0,55.0,54.0,60.0,54.0,277.0
4,20152,1,0,1,15835,6,TURMEQUE,1,I.E. TECNICA INDUSTRIAL,0.0,...,3.0,0.0,0.0,1.0,55.0,52.0,50.0,53.0,47.0,254.0
7,20172,1,0,2,15176,1,CHIQUINQUIRA,1,I.E. TECNICA PIO ALBERTO FERRO PEÑA,0.0,...,5.5,0.0,0.0,1.0,59.0,58.0,46.0,51.0,45.0,253.0
8,20172,1,0,2,15759,1,SOGAMOSO,0,COLEGIO COOPERATIVO REYES PATRIA,1.0,...,3.5,1.0,1.0,1.0,75.0,66.0,70.0,65.0,64.0,335.0


guardar archivos limpios

In [58]:
data_encoded.to_csv('data_encoded.csv', index=False)